docs/streams/developer-guide/dsl-api.html

```diff
@@ -3097,152 +3098,994 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
+            if (value == null) {
+                return null; // Skip null purchase amounts
+            }
+
+            // Get the current spending total for the customer
+            Double currentSpending = spendingStore.get(key);
+            if (currentSpending == null) {
+                currentSpending = 0.0;
+            }
+            // Update the cumulative spending
+            currentSpending += value;
+            spendingStore.put(key, currentSpending);
+
+            // Check if the customer qualifies for a discount
+            if (currentSpending &gt;= DISCOUNT_THRESHOLD) {
+                // Reset the spending after applying the discount
+                spendingStore.put(key, currentSpending - DISCOUNT_THRESHOLD);
+                // Return a notification message
+                return new KeyValue&lt;&gt;(key, DISCOUNT_NOTIFICATION_MESSAGE);
+            }
+            return null; // No discount, so no output for this record
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    private static class CumulativeDiscountProcessor implements Processor&lt;String, Double, String, String&gt; {
+        private KeyValueStore&lt;String, Double&gt; spendingStore;
+        private ProcessorContext&lt;String, String&gt; context;
+
+        @Override
+        public void init(final ProcessorContext&lt;String, String&gt; context) {
+            this.context = context;
+            // Retrieve the state store for cumulative spending
+            spendingStore = context.getStateStore(CUSTOMER_SPENDING_STORE);
+        }
+
+        @Override
+        public void process(final Record&lt;String, Double&gt; record) {
+            if (record.value() == null) {
+                return; // Skip null purchase amounts
+            }
+
+            // Get the current spending total for the customer
+            Double currentSpending = spendingStore.get(record.key());
+            if (currentSpending == null) {
+                currentSpending = 0.0;
+            }
+            // Update the cumulative spending
+            currentSpending += record.value();
+            spendingStore.put(record.key(), currentSpending);
+
+            // Check if the customer qualifies for a discount
+            if (currentSpending &gt;= DISCOUNT_THRESHOLD) {
+                // Reset the spending after applying the discount
+                spendingStore.put(record.key(), currentSpending - DISCOUNT_THRESHOLD);
+                // Send a discount notification
+                context.forward(record.withValue(DISCOUNT_NOTIFICATION_MESSAGE));
+            }
+        }
+    }
+}</code></pre>
+                <h5 id="categorizing-logs-by-severity-removal">Categorizing Logs by Severity</h5>
+                <p>Below, methods <code>categorizeWithFlatTransform</code> and <code>categorizeWithProcess</code> show how you can
+                    migrate from <code>flatTransform</code><br>to <code>process</code>.</p>
```

Comment:
Another `<br>` tag... Seems there is 18 overall that needs to get replace with a space

Reply:
Replaced.